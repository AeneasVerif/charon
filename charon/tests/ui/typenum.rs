//@ charon-args=--hide-marker-traits
pub struct UInt<U, B>(U, B);
pub struct UTerm;
pub struct B0;

pub type LongType = UInt<UInt<UInt<UInt<UInt<UInt<UTerm, B0>, B0>, B0>, B0>, B0>, B0>;

pub fn foo<T>() {}
pub fn main() {
    foo::<LongType>();
}
