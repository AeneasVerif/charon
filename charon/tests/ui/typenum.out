# Final LLBC before serialization:

// Full name: test_crate::UInt
pub struct UInt<U, B> {
  U,
  B,
}

// Full name: test_crate::UTerm
pub struct UTerm {}

// Full name: test_crate::B0
pub struct B0 {}

// Full name: test_crate::LongType
pub type LongType = UInt<UInt<UInt<UInt<UInt<UInt<UTerm, B0>, B0>, B0>, B0>, B0>, B0>

// Full name: test_crate::foo
pub fn foo<T>()
{
    let @0: (); // return

    @0 := ()
    @0 := ()
    return
}

// Full name: test_crate::main
pub fn main()
{
    let @0: (); // return
    let @1: (); // anonymous local

    storage_live(@1)
    @1 := foo<UInt<UInt<UInt<UInt<UInt<UInt<UTerm, B0>, B0>, B0>, B0>, B0>, B0>>()
    storage_dead(@1)
    @0 := ()
    @0 := ()
    return
}



